domain/ide/workspace/folder_test.go

```diff
@@ -534,7 +534,58 @@ func Test_processResults_ShouldCountSeverityByProduct(t *testing.T) {
 	f.processResults(scanData)
 
 	// Assert
-	require.Equal(t, 2, scanData.SeverityCount[product.ProductOpenSource][snyk.Critical.String()])
+	require.Equal(t, 2, scanData.SeverityCount[product.ProductOpenSource].Critical)
+}
+
+func Test_IncrementSeverityCount(t *testing.T) {
+	c := testutil.UnitTest(t)
+	c.SetAnalyticsEnabled(false)
+
+	engineMock, gafConfig := setUpEngineMock(t, c)
+
+	NewMockFolderWithScanNotifier(notification.NewNotifier())
+
+	issue := snyk.Issue{
+		Severity: snyk.Critical,
+		Product:  product.ProductOpenSource,
+	}
+
+	scanData := snyk.ScanData{
+		Product:       product.ProductOpenSource,
+		SeverityCount: make(map[product.Product]snyk.SeverityCount),
+		Issues:        []snyk.Issue{issue},
+	}
+
+	engineMock.EXPECT().GetConfiguration().AnyTimes().Return(gafConfig)
+	engineMock.EXPECT().InvokeWithInputAndConfig(localworkflows.WORKFLOWID_REPORT_ANALYTICS, gomock.Any(), gomock.Any()).Times(0)
+
+	// Act
+	incrementSeverityCount(&scanData, scanData.Issues[0])
+
+	// Assert
+	require.Equal(t, 1, scanData.SeverityCount[product.ProductOpenSource].Critical)
+}
+
+func Test_initializeSeverityCountForProductWhenScanDataIsEmpty(t *testing.T) {
+	c := testutil.UnitTest(t)
+
+	engineMock, gafConfig := setUpEngineMock(t, c)
+
+	NewMockFolderWithScanNotifier(notification.NewNotifier())
+
+	engineMock.EXPECT().GetConfiguration().AnyTimes().Return(gafConfig)
+	engineMock.EXPECT().InvokeWithInputAndConfig(localworkflows.WORKFLOWID_REPORT_ANALYTICS, gomock.Any(), gomock.Any()).Times(0)
+
+	scanData := snyk.ScanData{}
+
+	// Act
+	initializeSeverityCountForProduct(&scanData, "")
+
+	// Assert
+	require.Equal(t, 0, scanData.SeverityCount["unknown"].Critical)
+	require.Equal(t, 0, scanData.SeverityCount["unknown"].High)
+	require.Equal(t, 0, scanData.SeverityCount["unknown"].Medium)
+	require.Equal(t, 0, scanData.SeverityCount["unknown"].Low)
```

Comment:
I wanted to add this test:

```go
func Test_IncrementSeverityCountWhenIssueIsEmpty(t *testing.T) {
	c := testutil.UnitTest(t)
	c.SetAnalyticsEnabled(false)

	engineMock, gafConfig := setUpEngineMock(t, c)

	NewMockFolderWithScanNotifier(notification.NewNotifier())

	issue := snyk.Issue{}

	scanData := snyk.ScanData{
		SeverityCount: make(map[product.Product]snyk.SeverityCount),
		Issues:        []snyk.Issue{issue},
	}

	engineMock.EXPECT().GetConfiguration().AnyTimes().Return(gafConfig)
	engineMock.EXPECT().InvokeWithInputAndConfig(localworkflows.WORKFLOWID_REPORT_ANALYTICS, gomock.Any(), gomock.Any()).Times(0)

	// Act
	incrementSeverityCount(&scanData, scanData.Issues[0])

	// Assert
	require.Equal(t, 0, scanData.SeverityCount["unknown"].Critical)
}
```

And I thought the correct assertion was: `require.Equal(t, 0, scanData.SeverityCount["unknown"].Critical)`? But the output is `snyk.SeverityCount {Critical: 1, High: 0, Medium: 0, Low: 0}`.

The reason seems that since `Severity` is an "[enum](https://github.com/snyk/snyk-ls/blob/670ef46e4d5ec2f235ed0d0e9f69fe99c459798f/domain/snyk/issues.go#L188-L193)", when I'm passing an empty `Issue`, it gets initialised with its defaults values making `Critical = 0` which makes the `switch` to increment:


```go
	switch issue.Severity {
	case snyk.Critical:
		severityCount.Critical++ // <=== Here
	case snyk.High:
		severityCount.High++
	case snyk.Medium:
		severityCount.Medium++
	case snyk.Low:
		severityCount.Low++
	}
```

Could it be that this unit test is unreal?\\